flink-test-utils-parent/flink-migration-test-utils/src/main/java/org/apache/flink/test/migration/MigrationTestsSnapshotGenerator.java

```diff
@@ -0,0 +1,253 @@
+ * *(Test|ITCase)(.java|.scala)}. For scala tests, we also require the class name is the same with
+ * the containing filename.
+ */
+public class MigrationTestsSnapshotGenerator {
+
+    private static final Logger LOG =
+            LoggerFactory.getLogger(MigrationTestsSnapshotGenerator.class);
+
+    private static final String DEFAULT_PATH_PREFIXES = "src/test/java,src/test/scala";
+
+    private static final Option OPTION_HELP =
+            Option.builder()
+                    .longOpt("help")
+                    .required(false)
+                    .hasArg(false)
+                    .desc("print this help information.")
+                    .build();
+
+    private static final Option OPTION_DIR =
+            Option.builder()
+                    .longOpt("dir")
+                    .required()
+                    .hasArg()
+                    .desc("The root directory for scanning. Required.")
+                    .build();
+
+    private static final Option OPTION_VERSION =
+            Option.builder()
+                    .longOpt("version")
+                    .required()
+                    .hasArg()
+                    .desc("The version to generate. Required.")
+                    .build();
+
+    private static final Option OPTION_PREFIXES =
+            Option.builder()
+                    .longOpt("prefixes")
+                    .required(false)
+                    .hasArg()
+                    .desc(
+                            "The prefix paths to scan under the root directory, separated by \",\". Default to \""
+                                    + DEFAULT_PATH_PREFIXES
+                                    + "\"")
+                    .build();
+
+    private static final Option OPTION_CLASSES =
+            Option.builder()
+                    .longOpt("classes")
+                    .required(false)
+                    .hasArg()
+                    .desc(
+                            "The specified qualified class name to generate test data, "
+                                    + "separated by \",\". This option has a higher priority than the prefix option.")
+                    .build();
+
+    private static final Pattern VERSION_PATTERN = Pattern.compile("v?([0-9]+)[._]([0-9]+)");
+
+    private static final String CLASS_NAME_GROUP = "className";
+    private static final Pattern CLASS_NAME_PATTERN =
+            Pattern.compile("(?<" + CLASS_NAME_GROUP + ">[a-zA-Z0-9]*(Test|ITCase))(.java|.scala)");
+
+    public static void main(String[] args) {
+        for (String s : args) {
+            if (s.equals("--" + OPTION_HELP.getLongOpt())) {
+                HelpFormatter helpFormatter = new HelpFormatter();
+                helpFormatter.setOptionComparator(null);
+                helpFormatter.printHelp(
+                        "java " + MigrationTestsSnapshotGenerator.class.getName(), createOptions());
+                return;
+            }
+        }
```

Comment:
It might not work since during the parameter parsing, the commons-cli will look for the required parameters and throws exception if the required parameters is not found. 

I then added the explicit comments for this issue.

Reply:
ah ok, thanks for the clarification. could we add a comment explaining this to avoid others running into the same thing?